File: TP Evaluatorio_I_Aplicación_de_POO_en_otro_lenguaje._Hoy_tomamos_JAVA_y-PYTHON/C++/Class_concrect/Class_concrect.cpp
#include <iostream>
#include <fstream>
using namespace std;
class AudioPlayer
{
public:
AudioPlayer()
{
    cout << "The AudioPlayer constructor was invoked." << endl;
    isOpen = false;
    isPlaying = false;
    setVolume(10);
}
~AudioPlayer()
{
    cout << "The AudioPlayer destructor was invoked." << endl;
}
void open(string filePath)
{
    //@ToDo: por ahora simulamos la apertura correcta
    isOpen = true;
    cout << "The audiofile: " << filePath << " is open." << endl;
}
void play()
{
    //@ToDo: por ahora simulamos la reproducción correcta
    if(isOpen) isPlaying = true;
    cout << "The audiofile is playing." << endl;
}
void stop()
{
    //@ToDo: por ahora simulamos la detención correcta
    if(isPlaying) isPlaying = false;
    cout << "The audiofile is stopped." << endl;
}
void setVolume(float value)
{
    volume = value;
    cout << "The volume value is: " << volume << endl;
}
private:
    bool isOpen;
    bool isPlaying;
    float volume;
};

class VLC : public AudioPlayer
{
public:
VLC()
{
    cout << "The VLC constructor was invoked." << endl;
    setVolume(10);
    setPitch(0);
}
~VLC()
{
    cout << "The VLC destructor was invoked." << endl;
}
void setPitch(float value)
{
    pitch = value;
    cout << "The pitch value is: " << pitch << endl;
}
    float pitch;
};
int main()
{
    cout << endl;
AudioPlayer player;
    player.open("./resources/orchestral.ogg");

    player.play();

    player.setVolume(4);
    
    cout << endl;
VLC vlcPlayer;
    vlcPlayer.open("./resources/orchestral.ogg");

    vlcPlayer.play();

    vlcPlayer.setVolume(13);

    cout << endl;
    
    return EXIT_SUCCESS;
}

File: TP Evaluatorio_I_Aplicación_de_POO_en_otro_lenguaje._Hoy_tomamos_JAVA_y-PYTHON/Python/Class_concrect/Class_concrect.py
class AudioPlayer:

    

    def __init__(self):

        print("The AudioPlayer constructor was invoked.") 
        self.isOpen = False;
        self.isPlaying = False;
        self.setVolume(10);

    def __del__(self):

        print("The AudioPlayer destructor was invoked.")

    def open(self,filePath):

        #@ToDo: por ahora simulamos la apertura correcta
        self.isOpen = True;
        print("The audiofile: " + filePath + " is open.")

    def play(self):

        #@ToDo: por ahora simulamos la reproducción correcta
        if(self.isOpen) : self.isPlaying = True;
        print("The audiofile is playing.")

    def stop(self):

        #@ToDo: por ahora simulamos la detención correcta
        if(__isPlaying) : __isPlaying = False;
        print("The audiofile is stopped.")

    def setVolume(self, xd):

        self.volume = xd
        print("The volume value is: " + str(self.volume))


class VLC(AudioPlayer):

    def __init__(self):

        print("The VLC constructor was invoked.")
        self.setVolume(10);
        self.setPitch(0);

    def __del__(self):

        print("The VLC destructor was invoked.") 

    def setPitch(self, value):

        self.pitch = value;
        print("The pitch value is: " + str(self.pitch))

        

player = AudioPlayer()

player.open("./resources/orchestral.ogg");

player.play();

player.setVolume(4);

vlcPlayer = VLC()

vlcPlayer.open("./resources/orchestral.ogg");

vlcPlayer.play();

vlcPlayer.setVolume(13);

